problem-solver/cxx/uiModule/agents/JsonTranslateAgent.cpp

```diff
@@ -0,0 +1,314 @@
+/*
+* This source file is part of an OSTIS project. For the latest info, see http://ostis.net
+* Distributed under the MIT License
+* (See accompanying file COPYING.MIT or copy at http://opensource.org/licenses/MIT)
+*/
+
+#include "JsonTranslateAgent.hpp"
+#include <sc-memory/sc_link.hpp>
+#include <sc-memory/sc_memory.hpp>
+
+#include <sc-agents-common/utils/GenerationUtils.hpp>
+
+#include <sc-ui/uiPrecompiled.h>
+#include <sc-agents-common/utils/IteratorUtils.hpp>
+#include <sc-agents-common/utils/CommonUtils.hpp>
+#include <sc-agents-common/utils/AgentUtils.hpp>
+#include "keynodes/keynodes.hpp"
+
+using namespace std;
+using namespace utils;
+
+namespace uiModule {
+    StringStream ss;
+
+    SC_AGENT_IMPLEMENTATION(JsonTranslateAgent) {
+
+        if (!edgeAddr.IsValid())
+            return SC_RESULT_ERROR;
+        ScAddr questionNode = ms_context->GetEdgeTarget(edgeAddr);
+        ScAddr param = IteratorUtils::getFirstFromSet(ms_context.get(), questionNode);
+        if (!param.IsValid())
+            return SC_RESULT_ERROR_INVALID_PARAMS;
+        if (!componentIsValid(param)) {
+            SC_LOG_ERROR("Message processing program, component not found.")
+            utils::AgentUtils::finishAgentWork(ms_context.get(), param, false);
+            return SC_RESULT_ERROR;
+        }
+
+        ScAddr messageAddr;
+        try {
+            messageAddr = getTranslateJson(param);
+            generateMessage(messageAddr, param);
```

Comment:
This method can be removed. Just call `generateAnswer` to generate full construction